app/src/main/java/com/rizadwi/snapsift/common/base/BaseFragment.kt

```diff
@@ -0,0 +1,44 @@
+package com.rizadwi.snapsift.common.base
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.viewbinding.ViewBinding
+
+abstract class BaseFragment<T : ViewBinding> : Fragment() {
+
+    private var _binding: T? = null
+
+    val binding: T get() = _binding!!
+
+    abstract fun inflate(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): T
+
+    abstract fun setupView(view: View, savedInstanceState: Bundle?)
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = inflate(inflater, container, savedInstanceState)
+        return _binding!!.root
```

Comment:
don't use !!, change to be proper way to handle null value

Reply:
refactored in PR #15